src/audio/module_adapter/module_adapter.c

```diff
@@ -165,10 +166,11 @@ struct comp_dev *module_adapter_new_ext(const struct comp_driver *drv,
 err:
 #if CONFIG_IPC_MAJOR_4
 	if (mod)
-		rfree(mod->priv.cfg.input_pins);
-#endif
-	rfree(mod);
-	rfree(dev);
+		module_driver_heap_free(drv->user_heap, mod->priv.cfg.input_pins);
+#endif /* CONFIG_IPC_MAJOR_4 */
+
+	module_driver_heap_free(drv->user_heap, mod);
+	module_driver_heap_free(drv->user_heap, dev);
```

Comment:
this and a similar hunk below are unrelated bug-fixes, also fixed in https://github.com/thesofproject/sof/pull/10302/commits/0087192ae60e1857cd6cf4436113ca95cfddc050